src/components/NodeEditor/NodeEditor.tsx

```diff
@@ -228,12 +319,100 @@ export default class NodeEditor extends React.PureComponent<NodeEditorProps, Nod
         );
     }
 
+    private getInitialState(): NodeEditorState {
+        const type = this.determineConfigType();
+        const config = this.context.getTypeConfig(type);
+        const show = this.props.show || false;
+        const setResultName =
+            this.props.node.router && this.props.node.router.result_name ? true : false;
+
+        return {
+            config,
+            show,
+            setResultName
+        };
+    }
+
     private getLocalizedObject(): LocalizedObject {
         if (this.props.localizations && this.props.localizations.length) {
             return this.props.localizations[0];
         }
     }
 
+    private getResultNameField(): JSX.Element {
+        let nameField: JSX.Element;
+
+        if (this.state.setResultName) {
+            let resultName = '';
+
+            if (
+                (this.props.node.router as SwitchRouter) &&
+                (this.props.node.router as SwitchRouter).result_name
+            ) {
+                resultName += (this.props.node.router as SwitchRouter).result_name;
+            }
+
+            nameField = (
+                <TextInputElement
+                    data-spec="name-field"
+                    ref={this.onBindWidget}
+                    name="Result Name"
+                    showLabel={true}
+                    value={resultName}
+                    helpText="By naming the result, you can reference it later using @run.results.whatever_the_name_is"
+                    ComponentMap={this.props.ComponentMap}
+                    config={this.state.config}
+                />
+            );
+        } else {
+            nameField = (
+                <span
+                    data-spec="name-field"
+                    className={formStyles.saveLink}
+                    onClick={this.onShowNameField}>
+                    Save as..
+                </span>
+            );
+        }
+
+        return <div className={formStyles.saveAs}>{nameField}</div>;
+    }
+
+    private updateLocalizations(language: string, changes: LocalizationUpdates): void {
+        this.props.onUpdateLocalizations(language, changes);
+    }
+
+    /***
+     * If the user has a localized 'All Responses' case and they're adding another case,
+     * remove translation for the initial case.
+     * If the user is going from 1 or more cases to 0 and this router has a translation for the 'Other' case, lose it.
+     */
+    private cleanUpLocalizations(cases: CaseElementProps[]): void {
```

Comment:
this method ensures we're doing the right thing w/ existing localizations when the user updates a router; both `GroupRouter` and `SwitchRouter` call it in their `onValid` methods.

Reply:
this logic previously only existed in `SwitchRouter`